containers.go

```diff
@@ -164,17 +163,17 @@ type containerStore struct {
 func copyContainer(c *Container) *Container {
 	return &Container{
 		ID:             c.ID,
-		Names:          slices.Clone(c.Names),
+		Names:          copyStringSlice(c.Names),
 		ImageID:        c.ImageID,
 		LayerID:        c.LayerID,
 		Metadata:       c.Metadata,
-		BigDataNames:   slices.Clone(c.BigDataNames),
-		BigDataSizes:   maps.Clone(c.BigDataSizes),
-		BigDataDigests: maps.Clone(c.BigDataDigests),
+		BigDataNames:   copyStringSlice(c.BigDataNames),
+		BigDataSizes:   copyStringInt64Map(c.BigDataSizes),
+		BigDataDigests: copyStringDigestMap(c.BigDataDigests),
 		Created:        c.Created,
 		UIDMap:         copyIDMap(c.UIDMap),
 		GIDMap:         copyIDMap(c.GIDMap),
-		Flags:          maps.Clone(c.Flags),
+		Flags:          copyStringInterfaceMap(c.Flags),
```

Comment:
Are the maps changes actually an improvement? I think a copy of a value with `nil`s should again have `nil`s, not allocated empty maps.

(Applies throughout)

Reply:
Yes, you are right, but the previous behavior was that if the map was `nil`, it would return an empty map. This change hasn't led to a test failure yet, though, so we can use `maps.Clone` and hope that this change in behavior doesn't show up anywhere. WDYT?

Reply:
AFAICS none of these fields document any special semantics for `nil`, so callers should not be making any assumptions. Strictly speaking, that’s a bug in the callers, so we can do whatever we think best now that the question has come up.

I think it makes more sense to preserve `nil`ness:
- A “copy” changing that is just unexpected
- We do quite a lot of these copies (on every return from a c/storage API), so avoiding the allocations and GC cost seems desirable.